spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -256,6 +256,18 @@ final class TypeSig private(
     new TypeSig(initialTypes + dataType, maxAllowedDecimalPrecision, childTypes, litOnlyTypes,
       notes.+((dataType, note)))
 
+  /**
+   * Add a note about given types that marks them as partially supported.
+   * @param dataTypes the types this note is for.
+   * @param note the note itself
+   * @return the updated TypeSignature.
+   */
+  def withPsNote(dataTypes: List[TypeEnum.Value], note: String): TypeSig =
```

Comment:
Curious why `List` was chosen over `TypeEnum.ValueSet` or `Seq`?  Normally interfaces take `Seq` rather than `List` as many collection types are easily convertible to a sequence if they don't already provide that interface.